tests/api_tests.py

```diff
@@ -21,25 +21,20 @@ def test_device(self):
         populate_arena({'medium_motor' : [0, 'outA'], 'infrared_sensor' : [0, 'in1']})
 
         d = ev3.Device('tacho-motor', 'motor*')
-        self.assertTrue(d.connected)
 
         d = ev3.Device('tacho-motor', 'motor0')
-        self.assertTrue(d.connected)
 
         d = ev3.Device('tacho-motor', 'motor*', driver_name='lego-ev3-m-motor')
-        self.assertTrue(d.connected)
 
         d = ev3.Device('tacho-motor', 'motor*', address='outA')
-        self.assertTrue(d.connected)
 
-        d = ev3.Device('tacho-motor', 'motor*', address='outA', driver_name='not-valid')
-        self.assertTrue(not d.connected)
+        with self.assertRaises(ev3.DeviceNotFoundException):
+            d = ev3.Device('tacho-motor', 'motor*', address='outA', driver_name='not-valid')
 
         d = ev3.Device('lego-sensor', 'sensor*')
-        self.assertTrue(d.connected)
 
-        d = ev3.Device('this-does-not-exist')
-        self.assertFalse(d.connected)
+        with self.assertRaises(ev3.DeviceNotFoundException):
```

Comment:
same for these two

Reply:
other than that it looks good to me

Reply:
😟 Why didn't our tests catch this? That's... quite worrying.